app/src/main/java/org/fossasia/susi/ai/device/deviceconnect/DeviceConnectFragment.kt

```diff
@@ -86,6 +114,68 @@ class DeviceConnectFragment : Fragment(), IDeviceConnectView {
         context?.registerReceiver(receiverWifi, filter)
     }
 
+    override fun addDeviceProcess() {
+        room.visibility = View.GONE
+        addDeviceButton.visibility = View.VISIBLE
+        deviceList.visibility = View.VISIBLE
+        wifiList.visibility = View.VISIBLE
+
+        receiverWifi = WifiReceiver()
+        deviceConnectPresenter.searchDevices()
+
+        addDeviceButton.setOnClickListener {
+            deviceConnectPresenter.searchDevices()
+        }
+    }
+
+    // Function to show available rooms
+    override fun rooms() {
+        room.visibility = View.VISIBLE
+        addDeviceButton.visibility = View.GONE
+        deviceList.visibility = View.GONE
+        wifiList.visibility = View.GONE
+        showRooms()
+
+        add_room.setOnClickListener {
+            val roomName = edt_room.text.toString()
+            if (roomName != null) {
+                deviceConnectPresenter.addRoom(roomName)
+                Toast.makeText(context, "Added " + roomName + " as a room", Toast.LENGTH_SHORT).show()
+                edt_room.text.clear()
+                view?.let { it1 -> Utils.hideSoftKeyboard(context, it1) }
```

Comment:
To call the hideSoftKeyboard function we need to pass the view, so simply passing view from getview won't work because of the non-null assertion. So to avoid it, it's better to wrap with let.

Reply:
> What is view here? You can use rootView instead, no need for a null check.

Yes I know. Here `val rootView = inflater.inflate(R.layout.fragment_device_connect, container, false)` decleare `rootView` as lateinit var and use that. It won't be nullable.